data/migrations/V0073__update_ofec_committee_fulltext_and_candidate_fulltext_mvs.sql

```diff
@@ -0,0 +1,135 @@
+    COALESCE(totals.disbursements, (0)::numeric) AS disbursements,
+    COALESCE(totals.independent_expenditures, (0)::numeric) AS independent_expenditures,
+    COALESCE(totals.receipts, (0)::numeric) + COALESCE(totals.disbursements, (0)::numeric) + COALESCE(totals.independent_expenditures, (0)::numeric) AS total_activity
+   FROM ((ofec_committee_detail_mv cd
+     LEFT JOIN pacronyms pac USING (committee_id))
+     LEFT JOIN totals USING (committee_id));
+
+--Permissions-------------------
+
+ALTER TABLE ofec_committee_fulltext_mv_tmp OWNER TO fec;
+GRANT SELECT ON TABLE ofec_committee_fulltext_mv_tmp TO fec_read;
+GRANT ALL ON TABLE ofec_committee_fulltext_mv_tmp TO fec;
+
+--Indexes including unique idx--
+
+CREATE UNIQUE INDEX ofec_committee_fulltext_mv_idx_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING btree (idx);
+
+CREATE INDEX ofec_committee_fulltext_mv_fulltxt_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING gin (fulltxt);
+
+CREATE INDEX ofec_committee_fulltext_mv_receipts_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING btree (receipts);
+
+CREATE INDEX ofec_committee_fulltext_mv_disbursements_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING btree (disbursements);
+
+CREATE INDEX ofec_committee_fulltext_mv_independent_expenditures_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING btree (independent_expenditures);
+
+CREATE INDEX ofec_committee_fulltext_mv_total_activity_idx1_tmp ON ofec_committee_fulltext_mv_tmp USING btree (total_activity);
+
+----Rename view & indexes-------
+
+DROP MATERIALIZED VIEW IF EXISTS ofec_committee_fulltext_mv;
+
+ALTER MATERIALIZED VIEW IF EXISTS ofec_committee_fulltext_mv_tmp RENAME TO ofec_committee_fulltext_mv;
+
+SELECT rename_indexes('ofec_committee_fulltext_mv');
+
+/*------------------------------
+
+Candidates - add disbursements only
+
+-------------------------------*/
+
+DROP MATERIALIZED VIEW IF EXISTS ofec_candidate_fulltext_mv_tmp;
+
+CREATE MATERIALIZED VIEW ofec_candidate_fulltext_mv_tmp AS
+ WITH nicknames AS (
+         SELECT ofec_nicknames.candidate_id,
+            string_agg(ofec_nicknames.nickname, ' '::text) AS nicknames
+           FROM ofec_nicknames
+          GROUP BY ofec_nicknames.candidate_id
+        ), totals AS (
+         SELECT link.cand_id AS candidate_id,
+            sum(totals_1.receipts) AS receipts,
+            sum(totals_1.disbursements) AS disbursements
+           FROM (disclosure.cand_cmte_linkage link
+             JOIN ofec_totals_combined_mv totals_1 ON ((((link.cmte_id)::text = (totals_1.committee_id)::text) AND (link.fec_election_yr = (totals_1.cycle)::numeric))))
+          WHERE (((link.cmte_dsgn)::text = ANY ((ARRAY['P'::character varying, 'A'::character varying])::text[])) AND ((substr((link.cand_id)::text, 1, 1) = (link.cmte_tp)::text) OR ((link.cmte_tp)::text <> ALL ((ARRAY['P'::character varying, 'S'::character varying, 'H'::character varying])::text[]))))
+          GROUP BY link.cand_id
```

Comment:
This all looks okay if a bit explicit, which is normally a very good thing! :-)  With SQL, however, this can be a bit cumbersome to read, and as long as column types match not all of the explicit type referencing is necessary.  This sub-query, for instance, could be written a bit more concisely:

```sql
SELECT link.cand_id AS candidate_id, sum(totals_1.receipts) AS receipts, sum(totals_1.disbursements) AS disbursements
FROM disclosure.cand_cmte_linkage link
JOIN ofec_totals_combined_mv totals_1 ON link.cmte_id = totals_1.committee_id AND link.fec_election_yr = totals_1.cycle
WHERE link.cmte_dsgn IN ('P', 'A') AND (substr(link.cand_id, 1, 1) = link.cmte_tp OR link.cmte_tp NOT IN ('P', 'S', 'H'))
GROUP BY link.cand_id;
```

If you run `EXPLAIN ANALYZE` with both this version and the one written in the migration, the execution plan is exactly the same. :-)

I don't think it warrants a change here and it's tied to personal style, just something to note for the future!

Reply:
Thanks @ccostino! I inherited the `((((` here and in another issue but also found it tricky to read. I didn't want to mess with it but I appreciate knowing that it's not necessary. Thanks again!

Reply:
Thanks @ccostino  learn good style here:)
a minor question @lbeaufort , do you use a tool to generate this query? looks like very neat. Thanks

Reply:
@fec-jli I just copied the old MV over and tweaked it. Here's the original: https://github.com/fecgov/openFEC/blob/develop/data/migrations/V0026__more_public_schema_objects.sql#L1440

Reply:
I see. Thanks

Reply:
I wonder whether the original script was reverse engineered from `postgres` which is why it is explicit and verbose.

Reply:
I think you're right, @vrajmohan; I suspected the same thing when I saw the output of the query execution plan. :-)